OrrRock/OrrRock/SaveRootFinding/RouteFindingGymSaveViewController.swift

```diff
@@ -0,0 +1,309 @@
+        // 기기 대응한 테이블뷰셀의 개수
+        // SE 사이즈 - 2개 / 13 사이즈 - 3개 / max 사이즈 - 4개
+        maxTableViewCellCount = 1 + Int((UIScreen.main.bounds.height - 500) / 140)
+    }
+    
+    // 자동완성 헤더 레이블의 텍스트를 수정
+    func setTableViewHeaderLabel(text: String) {
+        tableViewHeaderLabel.text = text
+    }
+}
+
+//MARK: 오토레이아웃 설정 영역
+extension RouteFindingGymSaveViewController {
+    
+    func setUpLayout() {
+        view.addSubview(exitButton)
+        exitButton.snp.makeConstraints {
+            $0.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(16)
+            $0.leading.equalTo(view.safeAreaLayoutGuide.snp.leading).inset(16)
+            $0.height.equalTo(40)
+            $0.width.equalTo(40)
+        }
+        
+        view.addSubview(gymNameLabel)
+        gymNameLabel.snp.makeConstraints {
+            $0.leading.trailing.equalTo(view.safeAreaLayoutGuide).inset(OrrPd.pd16.rawValue)
+            $0.top.equalTo(exitButton).offset(OrrPd.pd72.rawValue)
+        }
+        
+        view.addSubview(gymTextField)
+        gymTextField.snp.makeConstraints{
+            $0.centerX.equalTo(view)
+            $0.top.equalTo(gymNameLabel.snp.bottom).offset(OrrPd.pd40.rawValue)
+            $0.horizontalEdges.equalTo(view).inset(OrrPd.pd16.rawValue)
+        }
+        
+        view.addSubview(nextButton)
+        nextButton.snp.makeConstraints {
+            $0.leading.trailing.equalToSuperview()
+            $0.bottom.equalTo(view.keyboardLayoutGuide.snp.top)
+            $0.height.equalTo(56)
+        }
+        
+        view.addSubview(autocompleteTableView)
+        autocompleteTableView.snp.makeConstraints {
+            $0.leading.trailing.equalToSuperview()
+            $0.bottom.equalTo(nextButton.snp.top)
+            $0.height.equalTo(50 * min(maxTableViewCellCount, filteredVisitedGymList.count))
+        }
+        
+        view.addSubview(tableViewHeaderLabel)
+        tableViewHeaderLabel.snp.makeConstraints {
+            $0.leading.equalToSuperview().offset(OrrPd.pd16.rawValue)
+            $0.trailing.equalToSuperview().offset(OrrPd.pd16.rawValue)
+            $0.bottom.equalTo(autocompleteTableView.snp.top).offset(-OrrPd.pd16.rawValue)
+        }
+    }
+        
+    // 자동완성 테이블 뷰의 데이터 개수의 변화에 따른 테이블뷰의 레이아웃의 변화가 필요한 경우에 본 함수를 호출
+    func resetAutocompleteTableView() {
+        autocompleteTableView.reloadData()
+        
+        autocompleteTableView.snp.removeConstraints()
+        autocompleteTableView.snp.makeConstraints {
+            $0.leading.trailing.equalToSuperview()
+            $0.bottom.equalTo(nextButton.snp.top)
+            $0.height.equalTo(50 * min(maxTableViewCellCount, filteredVisitedGymList.count))
+        }
+    }
+}
+
+extension RouteFindingGymSaveViewController: PHPickerViewControllerDelegate {
+    
+    func picker(_ picker: PHPickerViewController, didFinishPicking results: [PHPickerResult]) {
+        picker.dismiss(animated: true)
+        
+        var videoInfoArray : [VideoInfo] = []
```

Comment:
```suggestion
        var videoInformations : [VideoInfo] = []
```
p4.직관적인 변수명과 중복적인 네이밍을 피하기 위해 위와 같은 네이밍은 어떨까요?

Reply:
좋아요~!!
Swift 특성상 array 등의 타입명을 변수 명으로 사용하지 않더라구요!!
수정하는 편이 좋을 것 같습니다!!
감사합니다~